src/api/index.ts

```diff
@@ -2,8 +2,15 @@ import type { CatalogueItemType } from 'types/CatalogueItem.type'
 
 // eslint-disable-next-line import/prefer-default-export
 export const fetchCatalogItems = async (): Promise<CatalogueItemType[]> => {
-	const response = await fetch('api/data/catalog.json') // Replace with your own JSON file path
+	const response = await fetch('/api/data/catalog.json') // Replace with your own JSON file path
 	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 	const catalog: CatalogueItemType[] = await response.json()
 	return catalog
 }
+
+export const fetchCatalogItem = async (
+	catalogItemId?: string
+): Promise<CatalogueItemType | undefined> => {
+	const catalogItems: CatalogueItemType[] = await fetchCatalogItems()
+	return catalogItems.find(item => item.id === catalogItemId)
+}
```

Comment:
@butchtm I added a faux-API endpoint to get a specific catalog item given the id. Let me know what's your recommendation to handle this and filtering!

Reply:
Yes! this is exactly the same implem I'm going for.